src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,161 @@
+package duke;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.exceptions.DukeException1;
+
+/**
+ * Represents the hard disk/file that stores all the tasks in the application.
+ */
+public class Storage {
+    private String path;
+
+    /**
+     * Creates an object of the Storage class to store all tasks on the hard disk.
+     *
+     * @param filePath Path of the file that stores all the tasks in the application.
+     */
+    public Storage(String filePath) {
+        this.path = filePath;
+    }
+
+    /**
+     * Loads the tasks on the hard drive into an ArrayList when the application restarts.
+     *
+     * @return Arraylist containing all the tasks in the hard disk.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            String path = "data" + File.separator + "duke.txt";
+            boolean hasDirectory = Files.exists(Paths.get("data"));
+            boolean hasFile = Files.exists(Paths.get("data/duke.txt"));
+            if(hasDirectory) {
+                if(hasFile) {
+                    getContent("data/duke.txt", list);
+                    if(list.size() == 0) {
+                        System.out.println("There are no tasks");
+                    } else {
+                        //System.out.println("im here!!");
+                        printFile("data/duke.txt");
+                    }
+                } else {
+                    File file = new File(path);
+                }
+            } else {
+                File file = new File(path);
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("File not found");
+        } catch (IOException e) {
+            System.out.println("error!");
+        } finally {
+            return list;
+        }
+    }
+
+    /**
+     * Gets all the tasks in the hard disk
+     *
+     * @param filePath Path of the file that contains all the tasks in the application.
+     * @param list Arraylist that stores all the tasks.
+     * @throws FileNotFoundException If file is not found.
+     * @throws DukeException1 If there is an error in creating a todo, event, deadline task.
+     */
+    public void getContent(String filePath, ArrayList<Task> list) throws FileNotFoundException, DukeException1 {
```

Comment:
Try to avoid names with numbers (DukeException1) at the back, as stated [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/topics.html), under W4.6d.